src/main/scala/cromwell/binding/WdlBinding.scala

```diff
@@ -219,4 +221,28 @@ case class WdlBinding(ast: Ast) {
   }
 
   def findTask(name: String): Option[Task] = tasks.find(_.name == name)
+
+  /**
+   * Confirm all required inputs are present and attempt to coerce raw inputs to `WdlValue`s.
+   * This can fail if required raw inputs are missing or if the values for a specified raw input
+   * cannot be coerced to the target type of the input as specified in the binding.
+   */
+  def confirmAndCoerceRawInputs(rawInputs: Map[FullyQualifiedName, Any]): Try[Map[FullyQualifiedName, WdlValue]] = {
+    val tryCoercedValues = workflow.inputs.map { case (fqn, wdlType) =>
```

Comment:
I was thinking something along the lines of having separate composed functions, let me try that.